### PR TITLE
Get caption Functions

### PR DESCRIPTION
**Pull request by Humancyyborg**
- Randomly returns a string of text caption when called
Screen Shot 2021-04-05 at 2 05 03 PM

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/MemesServlet.java [hunk 1]
```diff
@@ -87,5 +92,40 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       String imgTag = String.format("<img src=\"%s\" width = \"300\"/>", blob.getMediaLink());
       response.getWriter().println(imgTag);
     }
+  }
+
+/** This funtions  randomly  get a string  captions from an array of string. When the function is called
```
Comment (ximenaparada) [inline src/main/java/com/google/sps/servlets/MemesServlet.java:92]: I like the descriptive blocks of comments and line comments you make before doing whatever it is you need to do.

File: src/main/java/com/google/sps/servlets/MemesServlet.java [hunk 2]
```diff
@@ -87,5 +92,40 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+ * it randomly return memes caption from an array
+ * input: null
+ * return: strings..
+ * 
+ */
+  
+
+  public static String getCpation(){
```
Comment (ximenaparada) [inline src/main/java/com/google/sps/servlets/MemesServlet.java]: I think that you should go through your code and fix spelling errors in method names or tags that you will call later on. [Like getCpation( ) --> getCaption( ) ] If it remains misspelled, other may have a hard time remembering or getting the method signature right when working with them. There are some words misspelled in the commenting but I think that's ok, just consider fixing the method names, variable names and such.
```suggestion
  public static String getCaption(){
```

File: src/main/java/com/google/sps/servlets/MemesServlet.java [hunk 3]
```diff
@@ -87,5 +92,40 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+    //   import the random module
+        Random random = new Random();
+        // get random  index of array element
+        int index = random.nextInt(proper_noun.length);
+       // System.out.println(proper_noun[index]);
+
+    //    return random caption
+        return proper_noun[index];
+        
+
```
Comment (ximenaparada) [inline src/main/java/com/google/sps/servlets/MemesServlet.java:123]: I don't know if it was your work, but lines 63 - 73 are a bit messy on the indentation (It's kind of in-line). It's a little confusing and hard to follow. Other than that, consider marking up the indentation a little bit more overall, I couldn't match all the brackets '{ }' to its pair as I was going through the code.

**Review by ximenaparada — CHANGES_REQUESTED** [review]
I think you did a good job! I'm looking forward to the more intricate implementation of your work and meme interaction.

File: src/main/java/com/google/sps/servlets/MemesServlet.java [hunk 4]
```diff
@@ -23,8 +23,13 @@
 @WebServlet("/memes")
 @MultipartConfig
 public class MemesServlet extends HttpServlet {
+<<<<<<< HEAD
   public final String PROJECT_ID = "spring21-sps-43";
   public final String BUCKET_NAME = "spring21-sps-43.appspot.com"
+=======
+  static final String PROJECT_ID = "spring21-sps-43";
+  static final String BUCKET_NAME = "spring21-sps-43.appspot.com";
+>>>>>>> ac2cd0ffba474995ca5e94bdfe941ce924b86195
```
Comment (renguyen) [inline src/main/java/com/google/sps/servlets/MemesServlet.java]: Remember to check "files changed" tab to make sure your changes are as intended :) This is an artifact from a merge conflict.
```suggestion
  public static final String PROJECT_ID = "spring21-sps-43";
  public static final String BUCKET_NAME = "spring21-sps-43.appspot.com";
```

File: src/main/java/com/google/sps/servlets/MemesServlet.java (hunk 2 above)
Comment (renguyen) [reply src/main/java/com/google/sps/servlets/MemesServlet.java]: +1 Good explanation, Ximena!

File: src/main/java/com/google/sps/servlets/MemesServlet.java [hunk 5]
```diff
@@ -87,5 +92,40 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+ * it randomly return memes caption from an array
+ * input: null
+ * return: strings..
+ * 
+ */
+  
+
+  public static String getCpation(){
+//     initilaise an array  of captions
+        final String[] proper_noun = {"Fred",
```
Comment (renguyen) [inline src/main/java/com/google/sps/servlets/MemesServlet.java]: nit: variable names in Java are typically lowerCamelCase, not separated by underscores. Also consider naming this more specifically, like `captions` or `captionsList`, instead of proper_noun.

File: src/main/java/com/google/sps/servlets/MemesServlet.java (hunk 2 above)
Comment (renguyen) [inline src/main/java/com/google/sps/servlets/MemesServlet.java]: Curious, why did you choose to make this "public static"? An alternative is making it a "private" helper function.

File: src/main/java/com/google/sps/servlets/MemesServlet.java (hunk 4 above)
Comment (ximenaparada) [reply src/main/java/com/google/sps/servlets/MemesServlet.java]: I was wondering what that was, first time seeing it.

**Review by ximenaparada — COMMENTED** [review]
It looks good! The renaming was addressed and I don't see anything worth noting other than good job. The comments explaining every step are helpful.

**Review by ximenaparada — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/MemesServlet.java [hunk 6]
```diff
@@ -87,5 +87,40 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+        "  When you’ve been eating healthy for the past 15 minutes and STILL see no progress.",
+
+        
+    
+    };
+    //   import the random module
+        Random random = new Random();
+        // get random  index of array element
+        int index = random.nextInt(captionList.length);
+       // System.out.println(proper_noun[index]);
```
Comment (renguyen) [inline src/main/java/com/google/sps/servlets/MemesServlet.java:118]: Next time, you can delete commented out code just to keep the main repo cleaner :)

**Review by renguyen — APPROVED** [review]
Approved!
